src/metabase/db/liquibase/h2.clj

```diff
@@ -6,6 +6,82 @@
    (liquibase.database.core H2Database)
    (liquibase.database.jvm JdbcConnection)))
 
+(defn create-udfs!
+  "Registers the JSON manipulation functions for an H2 database."
+  [^JdbcConnection conn]
+  (.execute (.createStatement conn) "
+CREATE ALIAS IF NOT EXISTS JSON_VALUE AS $$
+import org.json.simple.JSONObject;
+import org.json.simple.JSONValue;
+import org.json.simple.JSONArray;
+@CODE
+String jsonStringValue(String s, String key) {
+    Object obj = JSONValue.parse(s);
+    if (obj instanceof JSONObject) {
+        JSONObject jsonObject = (JSONObject) obj;
+        if (jsonObject.containsKey(key)) {
+            return jsonObject.get(key).toString();
+        } else {
+            return null;
+        }
+    } else if (obj instanceof JSONArray) {
+        JSONArray jsonArray = (JSONArray) obj;
+        int keyInt = Integer.parseInt(key);
+        if (keyInt < jsonArray.size() && keyInt >= 0) {
+            return jsonArray.get(keyInt).toString();
+        } else {
+            return null;
+        }
+    } else {
+        return null;
+    }
+}
+$$;")
```

Comment:
It might be possible to define these UDFs with Clojure, but I didn't want to bother with getting these to work nicely with the classloader.